pkg/controller/v1beta1/inferenceservice/reconcilers/ingress/ingress_reconciler.go

```diff
@@ -78,6 +79,16 @@ func getServiceHost(isvc *v1beta1.InferenceService) string {
 				1)
 		}
 	}
+	if isvc.Spec.Explainer != nil && deploymentMode == constants.ModelMeshDeployment {
+		if explainerStatus, ok := isvc.Status.Components[v1beta1.ExplainerComponent]; !ok {
+			return ""
+		} else if explainerStatus.URL == nil {
+			return ""
+		} else {
+			return strings.Replace(explainerStatus.URL.Host, fmt.Sprintf("-%s-default", string(constants.Explainer)), "",
+				1)
+		}
+	}
```

Comment:
I do not think we need this block as the `explain` verb matches the routing rule to explainer ksvc with the isvc host.

Reply:
This block is needed because in the case of explainer with ModelMesh predictor we don't want to return the ModelMesh serving url host, something like "modelmesh-serving.namespace:8033". The returned string will be added to hosts for the desiredIngress and fail the ingress creation. The logic could be implemented in other places. But it won't work without the logic of using explainer's url host instead predictor's when the explainer has ModelMesh predictor.

Reply:
We should really support ModelMesh ingress then as if only predictor presents the ingress should still work.

Reply:
I agree. That itself is a big work item. Unfortunately I can't address it in this PR.

Reply:
Can we add a @TODO here so we remember to address this in future?